part2/phonebook/src/App.js

```diff
@@ -0,0 +1,169 @@
+
+
+
+
+const App = () => {
+    const bodyStyle = {
+        fontFamily: `"Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
+            "Lucida Sans Unicode", Geneva, Verdana, sansSerif`
+    }
+
+    const h1Style = {
+        color: `#0c1616`,
+        textTransform: 'uppercase',
+        textAlign: 'center'
+    }
+
+    const h2Style = {
+        color: `#021818`,
+        textTransform: 'uppercase'
+    }
+
+    const [persons, setPersons] = useState([])
+    const [newName, setNewName] = useState('')
+    const [newNumber, setNewNumber] = useState('')
+    const [searchName, setSearchName] = useState('')
+    const [showAll, setShowAll] = useState(true)
+    const [message, setMessage] = useState(null)
+    const [error, setErrorMessage] = useState(null)
+
+
+    useEffect(() => {
+        personService
+            .getAll()
+            // https://javascript.info/promise-chaining
+            .then(initialData => {
+                setPersons(initialData)
+            })
+    }, [])
+
+
+
+    const addContact = (e) => {
+        e.preventDefault()
+
+        const newObject = {
+            name: newName,
+            number: newNumber
+        }
+
+
+        if (persons.some(person => person.name === newName)) {
+            const person = persons.find(p => p.name === newName)
+            const updatedPerson = { ...person, number: newNumber }
+
+            window.confirm(`${newName} is already in phonebook. Replace numbers?`)
+
+
+            personService
+                .update(updatedPerson.id, updatedPerson)
+                .then(() => {
+                    setPersons(persons.map(p => p.id !== updatedPerson.id ? p : updatedPerson))
+                    setNewName('')
+                    setNewNumber('')
+                    setMessage(`${newName}'s number is replaced`)
+                    setTimeout(() => {
+                        setMessage(null)
+                    }, 5000)
+                })
+
+                .catch(() => {
+                    // setMessage(null)
+                    setErrorMessage(`${newName} has allready been removed`)
+                    setTimeout(() => {
+                        setErrorMessage(null)
+                    }, 5000)
+                    setPersons(persons.filter(p => p.name !== newName))
+                }
+                )
+
+        } else {
+
+            personService
+                .create(newObject)
+                .then(returnedPerson => {
+                    setPersons(persons.concat(returnedPerson))
+                    setNewName('')
+                    setNewNumber('')
+                    setMessage(`${newName} is now in phonebook`)
+                    setTimeout(() => {
+                        setMessage(null)
+                    }, 5000)
+                })
+        }
+    }
+
+
+
+    const removePerson = (id) => {
+
+        const person = persons.find(p => p.id === id)
+        window.confirm(`Delete ${person.name}?`)
```

Comment:
![image](https://user-images.githubusercontent.com/1030080/98389810-3c42f300-205d-11eb-931c-64a8c1d2c4ea.png)

https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm

Šai funkcijai ir `boolean` return value, kuru tu ignorē. Un ja tu viņu ignorē, tad algoritms drošvien nedara to ko tu domā ko viņš dara. 😀